lang/cs/Org.Apache.REEF.IO/PartitionedData/Random/RandomInputPartition.cs

```diff
@@ -52,13 +53,18 @@ public string Id
             get { return _id; }
         }
 
-        public void Cache()
+        public CacheLevel Cache(CacheLevel level)
         {
             lock (_lock)
             {
+                if (level > CacheLevel.InMemoryMaterialized)
```

Comment:
I don't think we should, provided that we already return a `CacheLevel` indicating the lowest `CacheLevel`. The user can handle that logic on their own.